pkg/controllers/corecluster/corecluster_controller_test.go

```diff
@@ -0,0 +1,413 @@
+					Expect(testCoreCluster.Status.Conditions[0].Status).To(Equal(corev1.ConditionTrue))
+				})
+
+				It("should create the core cluster", func() {
+					By("without running the core cluster reconciliation")
+					coreCluster := capibuilder.Cluster().WithName(testInfraName).WithNamespace(testNamespaceName).Build()
+					Expect(cl.Create(ctx, coreCluster)).To(Succeed())
+
+					testCoreCluster := &capiv1.Cluster{}
+					Expect(cl.Get(ctx, client.ObjectKey{
+						Name:      testInfraName,
+						Namespace: testNamespaceName,
+					}, testCoreCluster)).To(Succeed())
+				})
+			})
+		})
+
+		Context("When there is an existing Core Cluster", func() {
+			BeforeEach(func() {
+				coreCluster = capibuilder.Cluster().WithNamespace(testNamespaceName).WithName(testInfraName).Build()
+
+				r = &CoreClusterReconciler{
+					ClusterOperatorStatusClient: operatorstatus.ClusterOperatorStatusClient{
+						Client:           cl,
+						ManagedNamespace: testNamespaceName,
+					},
+					Cluster:  coreCluster.DeepCopy(),
+					Infra:    openshiftInfrastructureCopy.DeepCopy(),
+					Platform: openshiftInfrastructureCopy.Status.PlatformStatus.Type,
+				}
+
+				Expect(cl.Create(ctx, coreCluster)).To(Succeed())
+			})
+
+			It("should update core cluster status", func() {
+				By("running the core cluster controller reconciliation")
+				_, err := r.Reconcile(ctx, reconcile.Request{
+					NamespacedName: types.NamespacedName{
+						Namespace: testNamespaceName,
+						Name:      testInfraName,
+					},
+				})
+				Expect(err).ToNot(HaveOccurred())
+
+				testCoreCluster := &capiv1.Cluster{}
+				Expect(cl.Get(ctx, client.ObjectKey{
+					Name:      testInfraName,
+					Namespace: testNamespaceName,
+				}, testCoreCluster)).To(Succeed())
+
+				Expect(testCoreCluster.Status.Conditions).ToNot(BeEmpty())
+				Expect(testCoreCluster.Status.Conditions[0].Type).To(Equal(capiv1.ControlPlaneInitializedCondition))
+				Expect(testCoreCluster.Status.Conditions[0].Status).To(Equal(corev1.ConditionTrue))
+			})
+
+			It("should update the core cluster status", func() {
+				By("without calling the core cluster reconciler")
+
+				updatedCoreCluster := coreCluster.DeepCopy()
+				conditions.MarkTrue(updatedCoreCluster, capiv1.ControlPlaneInitializedCondition)
+				Expect(cl.Status().Update(ctx, updatedCoreCluster)).To(Succeed())
+
+				testCoreCluster := &capiv1.Cluster{}
+				Expect(cl.Get(ctx, client.ObjectKey{
+					Name:      testInfraName,
+					Namespace: testNamespaceName,
+				}, testCoreCluster)).To(Succeed())
+
+				Expect(testCoreCluster.Status.Conditions).ToNot(BeEmpty())
+				Expect(testCoreCluster.Status.Conditions[0].Type).To(Equal(capiv1.ControlPlaneInitializedCondition))
+				Expect(testCoreCluster.Status.Conditions[0].Status).To(Equal(corev1.ConditionTrue))
+			})
+		})
+	})
+})
+
+var _ = Describe("Reconcile Core cluster on unsupported platform", func() {
```

Comment:
Ideally this could be a Context forking off in the main Describe rather than being its own one.

Reply:
Couldn't wrap my head around how to lay that out since the top describe's BeforeEach is creating an AWS cluster and for this context we want a cluster on an unsupported platform! Is there some way to clarify I want a cluster on a different platform than what's being created in the top BeforeEach?
When I had it in the same describe, it was failing, because it couldn't create the cluster on an unknown platform since an AWS one already existed :) but it's possible I was missing something and had it laid out incorrectly

Reply:
(we resolved this, there is in fact a tidy way to do this without two describes 🤓)